### DISCUSSION TITLE
How to correctly fix Package would be ignored

### DISCUSSION THREAD

**Discussion by JakobDev** [pypa/setuptools#3847] [Q&A]
During build I get this warning:
```
    ############################
    # Package would be ignored #
    ############################
    Python recognizes 'jdFlatpakSnapshot.translations' as an importable package,
    but it is not listed in the `packages` configuration of setuptools.

    'jdFlatpakSnapshot.translations' has been automatically added to the distribution only
    because it may contain data files, but this behavior is likely to change
    in future versions of setuptools (and therefore is considered deprecated).

    Please make sure that 'jdFlatpakSnapshot.translations' is included as a package by using
    the `packages` configuration field or the proper discovery methods
    (for example by using `find_namespace_packages(...)`/`find_namespace:`
    instead of `find_packages(...)`/`find:`).

    You can read more about "package discovery" and "data files" on setuptools
    documentation page.
```
together with
```
Installing 'jdFlatpakSnapshot.ui' as data is deprecated, please list it in `packages`
```

I was able to relove this, by adding
```toml
[tool.setuptools]
packages = ["jdFlatpakSnapshot", "jdFlatpakSnapshot.translations", "jdFlatpakSnapshot.ui"]
```
to my pyproject.toml. But don't think this is how it should be done. In my case, I have only a small Program, but I don't think this suitable for bigger ones. The Paths are also needed to be added to the MANIFEST.in. I also tried the other methods described in the [documentation](https://setuptools.pypa.io/en/latest/userguide/datafiles.html), but they seem to destroy the package structure.

My Programs have always a Directory named after the Program (jdFlatpakSnapshot in this case), which contains all the files.

**Comment by abravalheri** [comment]
Hi @JakobDev, what do you mean by "destroy the package structure"?

If you have a bigger package, you can use something like `[tool.setuptools.packages.find]\ninclude=["packagename*"]`.
If you prefer to have less configuration, you can switch to a [src-layout](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#src-layout), and omit `[tool.setuptools.packages]` completely.

Depending if you don't have any other folder in your package root other than the standard ones (e.g. `docs`, `tests` and the folder named after your project), you might also be able to leave `packages` unspecified in a [flat-layout](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#flat-layout) for setuptools to scan.

**Reply by JakobDev** [reply]
I have solved the problem by using
```toml
[tool.setuptools.package-dir]
jdFlatpakSnapshot = "jdFlatpakSnapshot"
```
This should be better documentated.

**Reply by abravalheri** [reply]
Hi @JakobDev, there is documentation about package discovery in https://setuptools.pypa.io/en/latest/userguide/package_discovery.html, but of course documentation improvements are always welcome. Please feel free to submit a PR with documentation improvements.

I am very curious to understand what do you mean by "destroy the package structure"?
